test/components/util/util.go

```diff
@@ -199,10 +210,60 @@ func WaitForDeployment(t *testing.T, client kubernetes.Interface, ns, name strin
 		t.Logf("all pods for deployment %s, are in ready state", deploy.Name)
 		return true, nil
 	}); err != nil {
+		if err := printPodLogs(t, client.CoreV1().Pods(ns), deploy.Spec.Selector); err != nil {
+			t.Logf("reading pod logs: %v", err)
+		}
+
 		t.Errorf("error while waiting for the pods: %v", err)
 	}
 }
 
+// printPodLogs print logs from all pods and containers selected by given selector.
+func printPodLogs(t *testing.T, p corev1typed.PodInterface, selector *metav1.LabelSelector) error {
```

Comment:
If this is called `printPodLogs`, then it should just print logs of a given pod. Not try to list pods (stress on the `s`).

I think the API should be concise where you just give podname and namespace and this function does the job of printing. Now if you want another function which given a label selector can print logs of all of them then create another one.

Reply:
I agree, but for the test code, that seems like a too much of an effort :smile: Maybe I can rename `printPodsLogs` ? Or maybe I should make it accept an array of pod names?

Reply:
Renamed to `PrintPodsLogs`, so it is also exported, so can be used in other tests.